### PR TITLE
@l2succes => Switch to use new unread methodology for the indicator

### PR DESCRIPTION
**Pull request by mzikherman**
And also to mark a message as being viewed.

This uses the branch in https://github.com/artsy/metaphysics/pull/873 , all the Impulse stuff is already on staging.

Tangentially, that Metaphysics PR also offers the `total_unread_count`, which precipitated this refactor.

#skip_new_tests

### CODE REVIEW

File: src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx [hunk 1]
```diff
@@ -158,9 +158,7 @@ export class ConversationSnippet extends React.Component<Props, any> {
             <TextPreview>
               <HorizontalLayout>
                 <SmallHeadline>{partnerName}</SmallHeadline>
-                <DateHeading>
-                  {conversation.is_last_message_to_user && !conversation.last_message_open && <UnreadIndicator />}
-                </DateHeading>
+                <DateHeading>{conversation.unread && <UnreadIndicator />}</DateHeading>
```
Comment (mzikherman) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: this is much nicer now

File: src/lib/Components/Inbox/Conversations/UpdateConversation.ts [hunk 2]
```diff
@@ -6,34 +6,33 @@ interface Conversation {
...
-  deliveryId: string,
+  fromLastViewedMessageId: string,
   onCompleted: MutationConfig<any>["onCompleted"],
   onError: MutationConfig<any>["onError"]
 ) {
   return commitMutation(environment, {
     updater: store => {
-      const currentTime = new Date().toISOString()
-      store.get(conversation.__id).setValue(currentTime, "last_message_open")
+      store.get(conversation.__id).setValue(false, "unread")
```
Comment (mzikherman) [inline src/lib/Components/Inbox/Conversations/UpdateConversation.ts]: pretty simple store update, the conversation is no longer unread.

**Comment by mzikherman** [comment]
Once https://github.com/artsy/metaphysics/pull/873 is merged I can undo the temp changes in here.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Inbox/Conversations/__tests__/UpdateConversation-tests.ts`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by mzikherman** [comment]
@l2succes I think this is good to go 💚 !

**Pull request merged**